presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlIntegrationSmokeTest.java

```diff
@@ -151,6 +151,30 @@ public void testMySqlTinyint1()
         assertUpdate("DROP TABLE mysql_test_tinyint1");
     }
 
+    @Test
+    public void testCharTrailingWhitespace()
+            throws Exception
+    {
+        execute("CREATE TABLE tpch.char_trailing_whitespace (x char(10))");
+        assertUpdate("INSERT INTO char_trailing_whitespace VALUES ('test')", 1);
+
+        // Test that the predicate doesn't need to be padded.
+        // Comparing to either the stripped or explicit value works.
+        assertQuery("SELECT * FROM char_trailing_whitespace WHERE x = 'test'", "VALUES 'test'");
+        assertQuery("SELECT * FROM char_trailing_whitespace WHERE x = 'test'", "VALUES 'test '");
+        assertQuery("SELECT * FROM char_trailing_whitespace WHERE x = 'test'", "VALUES 'test      '");
+
+        // Add additional trailing whitespace (and padding to full length) to query predicate.
+        assertQuery("SELECT * FROM char_trailing_whitespace WHERE x = 'test '", "VALUES 'test'");
+        assertQuery("SELECT * FROM char_trailing_whitespace WHERE x = 'test   '", "VALUES 'test'");
+        assertQuery("SELECT * FROM char_trailing_whitespace WHERE x = 'test      '", "VALUES 'test'");
```

Comment:
from testing perspective, there is no difference between 1 space, 3 spaces and 6 space. There should be following test cases:
- no trailing spaces
- some trailing spaces (eg 2 spaces)
- trailing spaces making it longer than char(10) (eg 8 spaces)
- leading spaces (these should not be ignored)

Also, the literal in the query must be explicitly of `char` type (`select typeof('a')` ⇒ `varchar(1)`), not `varchar`, to make sure the predicate pushdown can actually happen and uses the exact value you provided:
```
.... x = char 'test '
```